### PR TITLE
Add SQL server reference and minor refactor.

### PR DESCRIPTION
**Pull request by PeterJausovec**

### CODE REVIEW

File: src/command_modules/azure-cli-project/azure/cli/command_modules/project/custom.py [hunk 1]
```diff
@@ -382,18 +383,25 @@ def add_reference(target_group, target_name, reference_name):
...
 
         connection_string = results.connection_strings[0].connection_string
-        references.create_connection_string(
+        env_variables = references.create_documentdb_reference(
             service_name, reference_name, connection_string)
-        # Save the reference
-        project_settings.add_reference(
-            service_name, reference_name, instance.type)
-    else:
-        raise NotImplementedError()
```
Comment (philon-msft) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/custom.py]: Should we keep throwing this exception if the type isn't recognized?

File: src/command_modules/azure-cli-project/azure/cli/command_modules/project/references.py [hunk 2]
```diff
@@ -20,30 +22,70 @@ def get_secret_name(service_name, reference_name):
...
+               --from-literal={}={}\
+               --from-literal={}={}\
+               --from-literal={}={}'.format(secret_name,
+                                            admin_login_var, admin_login,
+                                            admin_password_var, admin_password,
+                                            fqdn_var, fqdn)
+    utils.execute_command(command)
+
+    # Label it with run=service_name
+    _label_secret(secret_name, 'run={}'.format(service_name))
```
Comment (philon-msft) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/references.py]: Should _label_secret always include the "run="? Or add a _label_run_secret helper to do it?

File: src/command_modules/azure-cli-project/azure/cli/command_modules/project/custom.py (hunk 1 above)
Comment (PeterJausovec) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/custom.py]: yep, adding it back!

File: src/command_modules/azure-cli-project/azure/cli/command_modules/project/references.py (hunk 2 above)
Comment (PeterJausovec) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/references.py]: will add a helper instead. I was thinking of hardcoding it, but I am not sure if we are planning to add more/different labels.

**Pull request merged**